src/compile/scale.ts

```diff
@@ -504,3 +506,78 @@ export function zero(scale: Scale, channel: Channel, fieldDef: FieldDef) {
   }
   return undefined;
 }
+
+/**
+ * Merge scales from children by unioning their domains
+ */
+export function mergeScales(model: LayerModel, channel: string) {
+  let scaleComp: ScaleComponents = null;
+
+  let domains = {};
+
+  model.children().forEach((child) => {
+    const childScales = child.component.scale[channel];
+    if (!childScales) {
+      return;
+    }
+
+    // range and other things will just be defined by the first scale
+    if (!scaleComp) {
+      scaleComp = childScales;
+    }
+
+    // the child domain may itself be a union of domains
+    const childDomain = childScales.main.domain;
+    if (isUnionedDomain(childDomain)) {
+      // if the domain is itself a union domain, concat
+      childDomain.fields.forEach((domain) => {
+        domains[hash(domain)] = domain;
+      });
+    } else {
+      domains[hash(childDomain)] = childDomain;
+    }
+
+    // add color legend and color legend bin scales if we don't have them yet
```

Comment:
Don't you have to check if the channel is `color`?   

Why should we run through these logic for merging other channels?